test/evaluateQueryNode.test.ts

```diff
@@ -178,14 +180,15 @@ const objects1InArrays = inArray(objects1)
 // object values actually have different keys. This tests a bit more stuff.
 
 const valuesForCategories: AnnotatedValue[][][] = [
-  [primitives, objects0, arrays, objects0InArrays],
-  [primitives, objects1, arrays, objects1InArrays],
+  [primitives, primitivesWithUnknown, objects0, arrays, objects0InArrays],
+  [primitives, primitivesWithUnknown, objects1, arrays, objects1InArrays],
 ]
 
 const SCHEMA: [] = []
 
 const ALL_CATEGORIES = [
   Category.PRIMITIVES,
+  Category.PRIMITIVES_WITH_UNKNOWN,
```

Comment:
Hmm. This means that when you use `ALL_CATEGORIES` then you actually get all of the primitives twice. I think we either need to:

1. Only have `PRIMITIVES_WITH_UNKNOWN`
2. Or have `PRIMITIVES` and `UNKNOWN`